### DISCUSSION TITLE
Cloning projects between instances using API

### DISCUSSION THREAD

**Discussion by mmmaxxxe** [directus/directus#16654] [Archive]
Hi all,
would it be possible to "clone" instances from a "master" project using API?

It will be a good point when cloning projects for example whose database engine is postgres to instances that uses mysql or sqlite or other.

Bye
M.

**Comment by azrikahar** [comment]
There isn't a direct "clone everything" endpoint via the API as of now. However, the general gist of "cloning" that can currently be achieved would be along the lines of:

1. using schema snapshot to snapshot/apply between the instances via CLI (it will be possible via API in the future with #14713)
2. making requests from master project (eg. get all permissions via `/permissions`), then making a POST request to the target instances with those permissions, but without their primary keys.

Though as the 2nd step outlined, this isn't a strict "clone" in the sense that the primary keys aren't preserved. (though I believe you can keep UUID primary keys when creating the items via the API)
